File: samples/JavaScript/text-processing-max-licenses-in-use.js
var file_system = new ActiveXObject("Scripting.FileSystemObject");
var fh = file_system.openTextFile('mlijobs.txt', 1); // 1 == open for reading
var in_use = 0, max_in_use = -1, max_in_use_at = [];

while ( ! fh.atEndOfStream) {
    var line = fh.readline();
    if (line.substr(8,3) == "OUT") {
        in_use++;
        if (in_use > max_in_use) {
            max_in_use = in_use;
            max_in_use_at = [ line.split(' ')[3] ];
        }
        else if (in_use == max_in_use)
            max_in_use_at.push( line.split(' ')[3] );
    }
    else if (line.substr(8,2) == "IN")
        in_use--;
}

fh.close();

WScript.echo("Max licenses out: " + max_in_use + "\n  " + max_in_use_at.join('\n  '));

File: samples/Go/text-processing-max-licenses-in-use.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "bytes"
)

var fn = "mlijobs.txt"

func main() {
    f, err := os.Open(fn)
    if err != nil {
        fmt.Println(err)
        return
    }
    defer f.Close()
    var ml, out int
    var mlTimes []string
    in := []byte("IN")
    for lr := bufio.NewReader(f); ; {
        line, pref, err := lr.ReadLine()
        if err == io.EOF {
            break
        }
        if err != nil {
            fmt.Println(err)
            return
        }
        if pref {
            fmt.Println("Unexpected long line.")
            return
        }
        f := bytes.Fields(line)
        if len(f) != 7 {
            fmt.Println("unexpected format,", len(f), "fields.")
            return
        }
        if bytes.Equal(f[1], in) {
            out--
            continue
        }
        out++
        if out == ml {
            mlTimes = append(mlTimes, string(f[3]))
            continue
        }
        if out > ml {
            ml = out
            mlTimes = append(mlTimes[:0], string(f[3]))
        }
    }
    fmt.Println("max licenses:", ml)
    fmt.Println("at:")
    for _, t := range mlTimes {
        fmt.Println(" ", t)
    }
}
